src/main/kotlin/carRacing/serviceimpl/DefaultCarFactory.kt

```diff
@@ -0,0 +1,26 @@
+package carRacing.serviceimpl
+
+import carRacing.domain.Car
+import carRacing.service.CarFactory
+import carRacing.service.MovementController
+import carRacing.view.CarInfo
+
+open class DefaultCarFactory : CarFactory {
+    override fun createCars(carCount: Int): List<Car> = throw NotImplementedError()
+
+    /*
+    * CarInfo의 movementController는 null이 넘어오지 않도록 DefaultCarFactory를 상속받는 클래스에서 구현해야 한다.
+    * Ex) RandomMovementCarFactory
+     */
+    override fun createCars(carInfoList: List<CarInfo>): List<Car> {
+        val carList = mutableListOf<Car>()
+        for (carInfo in carInfoList) {
+            val name: String = carInfo.name
+            val movementController: MovementController = carInfo.movementController!!
+
+            carList.add(Car(name, movementController))
+        }
+
+        return carList
+    }
```

Comment:
kotlin의 collections 확장함수를 활용해보면 아래와 같이 작성해볼 수 있어요 :)
```suggestion
    override fun createCars(carInfoList: List<CarInfo>): List<Car> {
        return carInfoList.map { info -> Car(name = info.name, info.movementController!!) }
    }
```